api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -411,12 +412,15 @@ public void setCruProxyAddresses(Collection<String> cruProxyAddresses) {
         this.cruProxyAddresses = cruProxyAddresses;
     }
 
-    public Collection<String> getCruPasswordHistory() {
-        return cruPasswordHistory;
+    public Set<String> getCruPasswordHistory() {
+        return ImmutableSet.copyOf(cruPasswordHistory);
```

Comment:
Is your returning of an `ImmutableSet` copy just an incidental better practice refactoring, or part of the required solution?